src/lib.rs

```diff
@@ -1028,6 +1015,48 @@ where
     }
 }
 
+#[cfg(not(feature = "httpcache"))]
+type HeaderValues = (Option<u32>, Option<u32>);
+#[cfg(feature = "httpcache")]
+type HeaderValues = (Option<u32>, Option<u32>, Option<Vec<u8>>);
+
+fn get_header_values(response: &Response<Body>) -> HeaderValues {
```

Comment:
one benefit if having this extracted from the execution path is that is now more easily unit testable. Would you be open to adding a unit test for `get_header_values`?

Reply:
done